eth-bytecode-db/eth-bytecode-db-server/src/services/database.rs

```diff
@@ -74,6 +75,27 @@ impl Database for DatabaseService {
         }))
     }
 
+    #[instrument(skip_all)]
+    async fn search_alliance_sources(
+        &self,
+        request: tonic::Request<SearchAllianceSourcesRequest>,
+    ) -> Result<tonic::Response<SearchSourcesResponse>, tonic::Status> {
+        let request = request.into_inner();
+        super::trace_request_metadata!(
+            chain_id = request.chain,
+            contract_address = request.address
+        );
+
+        let mut sources = vec![];
+        if let Some(alliance_db_client) = self.client.alliance_db_client.clone() {
+            sources = self
+                .search_alliance_sources(alliance_db_client, &request.chain, &request.address)
+                .await?;
+        }
```

Comment:
I'm not sure about this, but maybe we have to return 500 or 404 if `alliance_db_client` is not set..?

Reply:
Updated it to return `unavailable` for those cases, for now